esp-hal/src/dma/mod.rs

```diff
@@ -1704,7 +1733,11 @@ where
         buffer: &mut BUF,
     ) -> Result<(), DmaError> {
         let preparation = buffer.prepare();
-
+        #[cfg(esp32s3)]
+        if let Some(block_size) = preparation.block_size {
+            self.set_ext_mem_block_size(block_size.into());
+        }
```

Comment:
If PSRAM is used outside the esp32s3 an error should be returned here. (Ideally at compile time but that means more code :cry:)

Reply:
I'll return the error from new() - so it won't get this far.

Reply:
That won't work for custom implementations of the DmaTxBuffer trait and it won't work on the S2 and P4 where only some DMA channels support PSRAM.

Reply:
ok - I'll check it here too.